### DISCUSSION TITLE
How do I programatically preload client data into a route using Link or redirect?

### DISCUSSION THREAD

**Discussion by jscul** [remix-run/remix#7092] [Q&A] [answered]
I want to click a `<Link>` and automatically get put on that page regardless of whether `loader` has completed or not. I know I can use `useFetcher` for this, but that makes it so SSR no longer works. I can't use `shouldRevalidate` because I'm navigating from a different route so it doesn't get called.

This seems like it'd be a pretty common use case... I already have _some_ data loaded, when I navigate to a new page I want to prefill that page immediately while I wait for `loader` to do it's thing... but still have SSR.

Is the only way to accomplish this through 2 different URL's, one that's SSR and one that's not?

**Comment by kiliman** [comment, marked as answer]
You should checkout `defer`.

https://remix.run/docs/en/1.19.2/guides/streaming#using-defer

**Reply by jscul** [reply]
This is exactly what I was looking for, would you happen to know if there's a way to determine if we're dealing with an initial page load versus a client navigation in a loader?
